File: app/src/main/java/com/startogamu/zikobot/module/soundcloud/model/SoundCloudPlaylist.java

package com.startogamu.zikobot.module.soundcloud.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import org.parceler.Parcel;

import java.util.ArrayList;
import java.util.List;

import lombok.Data;

@Parcel
@Data
public class SoundCloudPlaylist {

    @SerializedName("kind")
    @Expose
    public String kind;
    @SerializedName("id")
    @Expose
    public Integer id;
    @SerializedName("created_at")
    @Expose
    public String createdAt;
    @SerializedName("user_id")
    @Expose
    public Integer userId;
    @SerializedName("duration")
    @Expose
    public Integer duration;
    @SerializedName("sharing")
    @Expose
    public String sharing;
    @SerializedName("tag_list")
    @Expose
    public String tagList;
    @SerializedName("permalink")
    @Expose
    public String permalink;
    @SerializedName("track_count")
    @Expose
    public Integer trackCount;
    @SerializedName("streamable")
    @Expose
    public Boolean streamable;
    @SerializedName("downloadable")
    @Expose
    public Boolean downloadable;
    @SerializedName("embeddable_by")
    @Expose
    public String embeddableBy;
    @SerializedName("type")
    @Expose
    public String type;
    @SerializedName("playlist_type")
    @Expose
    public String playlistType;
    @SerializedName("ean")
    @Expose
    public String ean;
    @SerializedName("description")
    @Expose
    public String description;
    @SerializedName("genre")
    @Expose
    public String genre;
    @SerializedName("release")
    @Expose
    public String release;
    @SerializedName("label_name")
    @Expose
    public String labelName;
    @SerializedName("title")
    @Expose
    public String title;
    @SerializedName("license")
    @Expose
    public String license;
    @SerializedName("uri")
    @Expose
    public String uri;
    @SerializedName("permalink_url")
    @Expose
    public String permalinkUrl;
    @SerializedName("artwork_url")
    @Expose
    public String artworkUrl;
    @SerializedName("user")
    @Expose
    public User user;
    @SerializedName("tracks")
    @Expose
    public List<SoundCloudTrack> soundCloudTracks = new ArrayList<SoundCloudTrack>();

}

File: data/src/main/java/com/joxad/zikobot/data/module/soundcloud/model/SoundCloudPlaylist.kt
package com.joxad.zikobot.data.module.soundcloud.model

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName


import java.util.ArrayList

import lombok.Data

class SoundCloudPlaylist {

    @SerializedName("kind")
    @Expose
    var kind: String? = null
    @SerializedName("id")
    @Expose
    var id: Int? = null
    @SerializedName("created_at")
    @Expose
    var createdAt: String? = null
    @SerializedName("user_id")
    @Expose
    var userId: Int? = null
    @SerializedName("duration")
    @Expose
    var duration: Int? = null
    @SerializedName("sharing")
    @Expose
    var sharing: String? = null
    @SerializedName("tag_list")
    @Expose
    var tagList: String? = null
    @SerializedName("permalink")
    @Expose
    var permalink: String? = null
    @SerializedName("track_count")
    @Expose
    var trackCount: Int? = null
    @SerializedName("streamable")
    @Expose
    var streamable: Boolean? = null
    @SerializedName("downloadable")
    @Expose
    var downloadable: Boolean? = null
    @SerializedName("embeddable_by")
    @Expose
    var embeddableBy: String? = null
    @SerializedName("type")
    @Expose
    var type: String? = null
    @SerializedName("playlist_type")
    @Expose
    var playlistType: String? = null
    @SerializedName("ean")
    @Expose
    var ean: String? = null
    @SerializedName("description")
    @Expose
    var description: String? = null
    @SerializedName("genre")
    @Expose
    var genre: String? = null
    @SerializedName("release")
    @Expose
    var release: String? = null
    @SerializedName("label_name")
    @Expose
    var labelName: String? = null
    @SerializedName("title")
    @Expose
    var title: String? = null
    @SerializedName("license")
    @Expose
    var license: String? = null
    @SerializedName("uri")
    @Expose
    var uri: String? = null
    @SerializedName("permalink_url")
    @Expose
    var permalinkUrl: String? = null
    @SerializedName("artwork_url")
    @Expose
    var artworkUrl: String? = null
    @SerializedName("user")
    @Expose
    var user: SoundCloudUser? = null
    @SerializedName("tracks")
    @Expose
    var soundCloudTracks: List<SoundCloudTrack> = ArrayList()

    companion object {

        fun favorite(user: SoundCloudUser, soundCloudTracks: ArrayList<SoundCloudTrack>): SoundCloudPlaylist {
            val soundCloudPlaylist = SoundCloudPlaylist()
            soundCloudPlaylist.user =(user)
            soundCloudPlaylist.title = "Favoris"
            soundCloudPlaylist.artworkUrl = user.avatarUrl
            soundCloudPlaylist.soundCloudTracks =(soundCloudTracks)
            soundCloudPlaylist.trackCount = (soundCloudTracks.size)
            return soundCloudPlaylist
        }
    }
}
